src/components/RadioButton/RadioButton.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { forwardRef } from 'react'
+import styled, { css, down, typography } from '@xstyled/styled-components'
+
+import { Typography } from '../'
+import { Icon } from '../Iconography'
+
+const RadioButton = forwardRef(({ color, label, ...props }, ref) => {
+  return (
+    <LabelContainer color={color}>
+      <Input type='radio' ref={ref} {...props} />
+      <CheckedIcon icon='radio_button_checked' color='primary' />
+      <UncheckedIcon icon='radio_button_outline' />
+      {label && (
+        <Typography fontSize={3} lineHeight={3} fontWeight={0} paddingLeft={3}>
+          {label}
+        </Typography>
+      )}
+    </LabelContainer>
+  )
+})
+
+const CheckedIcon = styled(Icon)`
+  display: none;
+`
+const UncheckedIcon = styled(Icon)`
+  & path {
+    fill: gray.700;
+  }
+`
+const LabelContainer = styled.label`
+  display: inline-flex;
+  vertical-align: top;
+  position: relative;
+  user-select: none;
+  color: gray.800;
+  min-height: 24px;
+
+  ${down(
+    'sm',
+    css`
+      display: flex;
```

Comment:
por que precisou alterar pra flex aqui?

Reply:
Para aquela função de, caso seja mostrado no mobile, ele mostre um radio button por linha.

Reply:
Tu diz quem for usar o componente? Se for isso, concordo, faz sentido.